python/ql/test/experimental/query-tests/Security/CWE-208/PossibleTimingAttackAgainstHash.qlref

```diff
@@ -0,0 +1 @@
+experimental/Security/CWE/CWE-208/PossibleTimingAttackAgainstHash.ql
```

Comment:
You are missing a subdirectory (`TimingAttackAgainstHash`) from this reference, and hence the tests don't currently pass.

In total, three tests are currently failing because of this: 
```
156 tests passed; 3 tests failed:
  FAILED: /home/runner/work/semmle-code/semmle-code/ql/python/ql/test/experimental/query-tests/Security/CWE-208/PossibleTimingAttackAgainstHash.qlref
  FAILED: /home/runner/work/semmle-code/semmle-code/ql/python/ql/test/experimental/query-tests/Security/CWE-208/TimingAttackAgainstSensitiveInfo.qlref
  FAILED: /home/runner/work/semmle-code/semmle-code/ql/python/ql/test/experimental/query-tests/Security/CWE-208/TimingAttackAgainstHeaderValue.qlref
```

Reply:
Sorry i didn't notice.

Reply:
No worries. :slightly_smiling_face:
I'll start the tests again.

Reply:
We're getting closer. Now the tests are compiling correctly, but some of them are failing due to missing `.expected` files`:

```
  FAILED: /home/runner/work/semmle-code/semmle-code/ql/python/ql/test/experimental/query-tests/Security/CWE-208/TimingAttackAgainstHeaderValue.qlref
  FAILED: /home/runner/work/semmle-code/semmle-code/ql/python/ql/test/experimental/query-tests/Security/CWE-208/PossibleTimingAttackAgainstSensitiveInfo.qlref
```

Specifically:
```
Cannot find TimingAttackAgainstHeaderValue.expected file.
Cannot find PossibleTimingAttackAgainstSensitiveInfo.expected file.
```

Reply:
Hi @tausbn,  why the test fails again ?

Reply:
One of our CI checks is that all QL code must be autoformatted (which can be done via the VSCode extension, or through the CLI using `codeql query format`).

Specifically, the following files need to be autoformatted:
```
Error: ql/python/ql/src/experimental/semmle/python/security/TimingAttack.qll would change by autoformatting.
Error: ql/python/ql/src/experimental/Security/CWE-208/TimingAttackAgainstHash/TimingAttackAgainstHash.ql would change by autoformatting.
Error: ql/python/ql/src/experimental/Security/CWE-208/TimingAttackAgainstSensitiveInfo/TimingAttackAgainstSensitiveInfo.ql would change by autoformatting.
Error: ql/python/ql/src/experimental/Security/CWE-208/TimingAttackAgainstHeaderValue/TimingAttackAgainstHeaderValue.ql would change by autoformatting.
```

Also, the following warnings were emitted, which should also be addressed:
```
WARNING: The @id property should be a valid query identifier. (/home/runner/work/semmle-code/semmle-code/ql/python/ql/src/experimental/Security/CWE-208/TimingAttackAgainstHash/PossibleTimingAttackAgainstHash.ql:1,1-13,4)
Warning: WARNING: A path-problem query must select a string in its 6th column. (/home/runner/work/semmle-code/semmle-code/ql/python/ql/src/experimental/Security/CWE-208/TimingAttackAgainstHash/PossibleTimingAttackAgainstHash.ql:37,3-19)
```
(For the first one, I believe the issue is the upper case `H` in `Hash`. Replacing it with `h` should make the warning go away.)

Finally, there are also some issues in the `.qhelp` files (but these are less critical):

```
[ERROR] generate query-help> /home/runner/work/semmle-code/semmle-code/ql/python/ql/src/experimental/Security/CWE-208/TimingAttackAgainstHash/TimingAttackAgainstHash.qhelp: Could not find sample UnsafeComparisonOfHash.py
[ERROR] generate query-help> /home/runner/work/semmle-code/semmle-code/ql/python/ql/src/experimental/Security/CWE-208/TimingAttackAgainstSensitiveInfo/TimingAttackAgainstSensitiveInfo.qhelp: Could not find sample UnsafeComparisonOfSensitiveInfo.py
```
The fix for these should be fairly simple. You just need to update the files to point at the correct file name for the sample code.